File: non_native/lib/model/bird.dart
import 'habitat.dart';

class Bird {
  final int id;
  final String name;
  final String order;
  final String family;
  final Habitat habitat;
  final int sightCount;

  Bird({
    required this.id,
    required this.name,
    required this.order,
    required this.family,
    required this.habitat,
    required this.sightCount,
  });

  factory Bird.fromJson(Map<String, dynamic> json) {
    return Bird(
      id: json['id'] as int,
      name: json['name'] as String,
      order: json['order'] as String,
      family: json['family'] as String,
      habitat: HabitatExtension.fromString(json['habitat'] as String),
      sightCount: json['sightCount'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'order': order,
      'family': family,
      'habitat': habitat.toJson(),
      'sightCount': sightCount,
    };
  }
}

File: native_server/app/src/main/java/com/example/myapp/birds/domain/model/Bird.kt
package com.example.myapp.birds.domain.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
@Entity(tableName = "birds")
data class Bird(
    @PrimaryKey()
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val order: String,
    val family: String,
    val habitat: Habitat,
    val sightCount: Int
) {
}